packages/shared/src/components/post/PostModalActions.tsx

```diff
@@ -78,6 +84,27 @@ export function PostModalActions({
     }
   };
 
+  const promotePostPrompt = async () => {
+    const promoteFlag = post.flags?.promoteToPublic;
+
+    const options: PromptOptions = {
+      title: promoteFlag ? 'Demote post' : 'Promote post',
+      description: `Do you want to ${
+        promoteFlag ? 'demote' : 'promote'
+      } this post ${promoteFlag ? 'from' : 'to'} the public?`,
+      okButton: {
+        title: promoteFlag ? 'Demote' : 'Promote',
+      },
+    };
+    if (await showPrompt(options)) {
+      if (promoteFlag) {
+        await demotePost(post.id);
+      } else {
+        await promotePost(post.id);
+      }
+    }
```

Comment:
Should we not always try to utilize calling requests through `useMutation`?

Reply:
I think for these ones its fine, only ido uses them, and no problem  if they have simple requests